### PR TITLE
Modernize NO3/NP3 Bone Scimitar

### PR DESCRIPTION
**Pull request by bismurphy**
NO3 and NP3 had a weird mix of Bone Scimitar functions where each was partially complete and missing different parts.

Now both are complete and modernized. Could use another pass to make it PSP-matching, but for now I'll leave this one at this point where the code seems nice.

### CODE REVIEW

File: src/st/no3/e_bone_scimitar.c [hunk 1]
```diff
@@ -1,35 +1,276 @@
...
+
+        if (self->animFrameIdx == 7 && self->animFrameDuration == 0) {
+            PlaySfxPositional(SFX_BONE_SWORD_SWISH_B);
+        }
+
+        if (animStatus == 0) {
+            SetStep(BONE_SCIMITAR_WALK_AWAY_FROM_PLAYER);
+            self->ext.boneScimitar.attackTimer =
+                D_80183C18[self->params % 2]
+                          [(++self->ext.boneScimitar.attackTimerIndex) & 3];
```
Comment (hohle) [inline src/st/no3/e_bone_scimitar.c:146]: This looks like it was copied from the original, but it's s a bit terse for my taste. The increment of `attackTimerIndex` isn't related to the operation of looking up and assigning the attack timer, so my preference would be that they were separated.

I'd rather get the coverage, however, so if nothing else is blocking feel free to ignore.

**Review by hohle — APPROVED** [review]

File: src/st/no3/e_bone_scimitar.c (hunk 1 above)
Comment (bismurphy) [reply src/st/no3/e_bone_scimitar.c:146]: Good call, agreed! Updated to split the increment and access into two lines.

**Comment by sozud** [comment]
Looks good, needs a rebase to account for the dedupe PR

**Comment by bismurphy** [comment]
rebase got messed up, fixing

**Review by sozud — APPROVED** [review]

**Pull request merged**